### DISCUSSION TITLE
A better way to visualize Hexagonal Architecture?

### DISCUSSION THREAD

**Discussion by valentinajemuovic** [techexcellenceio/techexcellence#20] [Ideas]
A common question is understanding Hexagonal Architecture, and the diagram with the concentric circles may be too confusing.
The reason is due to differences in static dependencies, versus the runtime flow.

Recording here an idea from @peterschuler about visualizing it as an animation.

I think this would also be useful for the Tech Excellence community page, as a quick way to represent Hexagonal Architecture because it's a really common theme that comes up

Perhaps also visualizations for unit tests and test robustness?

FYI @nicolasrosado

**Comment by peterschuler** [comment]
I also feel that when you move from a 3 layered architecture the hexagonal architecture looks really complex and big. While all it does is pull the middle layer to the top of the pyramid and flip one dependency arrow (the one between the service and the data layer).

Come to think of it... a 3D concentric model might even be better.. and much cooler looking.. but not really making the transition easier.
